### ISSUE TITLE
Add support for brotli compression to Web Cmdeltes

### ISSUE THREAD

**Issue by markekraus** [PowerShell/PowerShell#4947]
[brotli](https://en.wikipedia.org/wiki/Brotli) will likely become the compressed transfer encoding of choice. `Invoke-Webrequest` and `Invoke-RestMethod` should support these natively. 

Steps to reproduce
------------------

```powershell
Invoke-WebRequest https://www.bayden.com/test/brotliimg.aspx -Headers @{'Accept-Encoding' =
 'br'} -OutFile C:\temp\brotli.jpg
```

Expected behavior
-----------------
`C:\temp\brotli.jpg` should be a jpeg

Actual behavior
---------------
`C:\temp\brotli.jpg` is brotli compressed data.

Environment data
----------------

```none
Name                           Value
----                           -----
PSVersion                      6.0.0-beta
PSEdition                      Core
GitCommitId                    v6.0.0-beta.7
OS                             Microsoft Windows 10.0.15063
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0

```

**Comment by iSazonov** [comment]
Related blog post https://blogs.msdn.microsoft.com/dotnet/2017/07/27/introducing-support-for-brotli-compression/

**Comment by markekraus** [comment]
I'm not sure if there are plans to add automated Brotli decompression to `HttpClientHandler`. I dug around and couldn't find anything for it. It may need to be handled after the response is returned.

**Comment by iSazonov** [comment]
Could you please create Issue-Question-Request in CoreFX repo?

**Comment by SteveL-MSFT** [comment]
Looks like it's coming at some time in CoreFx https://github.com/dotnet/corefx/issues/20558#issuecomment-314177535

**Comment by iSazonov** [comment]
Good news!
Question is do we add the feature today by means of third-party nugget package?

**Comment by markekraus** [comment]
I'm probably the least qualified to say this, but my gut tells me it's better to wait until it is "native". If not due to my trust issues with compression libraries, then at least for the refactor effort. I think it makes a good 6.1.0 goal. Most hosts will probably still fallback to gzip or identity in the time it takes for this to become native which looks likely to be well before 6..1.0

**Comment by iSazonov** [comment]
I agree that best way is to follow CoreFX. The only question is how good to have it now.

**Comment by markekraus** [comment]
From what little I can gather, brotli doesn't have mass adoption yet and just about any site that does impliment it currently also has a gzip fallback. Running into a site that is pure brotli without at least identity fallback (outside of the scope of sites dedicated to testing brotli) are outliers. I think the convenience of having it does not trump waiting for the native solution and that will likely not lead to any negative user experience. In the outliers who require it, they can still post-process their response, so they are not without remedy, only without convenience.

**Comment by iSazonov** [comment]
@SteveL-MSFT Could you assign a milestone if it is approved?

**Comment by markekraus** [comment]
adding  dotnet/corefx#25785 as a reference.

**Comment by SteveL-MSFT** [comment]
Latest nightly w/ merged dotnetcore2.1-rc1 include System.Compression.Brotli

**Comment by JustinGrote** [comment]
@SteveL-MSFT I'd be willing to attempt a PR as a httpclient Middleware to look for brotli compression and transparently decode it if there is interest, should be no breaking behavior

**Comment by iSazonov** [comment]
Fixed in #17955
